### PR TITLE
feat: Rest Api for did handlers

### LINKED ISSUE #1570: Rest Api for did handlers

### PR DESCRIPTION
**Pull request by talwinder50**
closes #1570

Signed-off-by: talwinder.kaur <talwinder.kaur@securekey.com>

### CODE REVIEW

File: pkg/controller/rest/vdri/operation.go [hunk 1]
```diff
@@ -81,6 +91,49 @@ func (o *Operation) CreatePublicDID(rw http.ResponseWriter, req *http.Request) {
 	rest.Execute(o.command.CreatePublicDID, rw, bytes.NewReader(reqBytes))
 }
 
+// SaveDID swagger:route POST /vdri/did vdri saveDIDReq
+//
+// Saves a did document with the friendly name...
```
Comment (rolsonquadras) [inline pkg/controller/rest/vdri/operation.go:96]: extra dots at the end :)

File: pkg/controller/rest/vdri/operation_test.go [hunk 2]
```diff
@@ -8,13 +8,17 @@ package vdri
...
+	"encoding/base64"
 	"encoding/json"
 	"fmt"
 	"io"
 	"net/http"
 	"net/http/httptest"
 	"testing"
 
+	mockprovider "github.com/hyperledger/aries-framework-go/pkg/internal/mock/provider"
+	mockstore "github.com/hyperledger/aries-framework-go/pkg/mock/storage"
```
Comment (rolsonquadras) [inline pkg/controller/rest/vdri/operation_test.go:20]: These should be grouped with other aries-framework-go imports.

File: pkg/controller/rest/vdri/operation_test.go [hunk 3]
```diff
@@ -87,6 +135,148 @@ func TestOperation_CreatePublicDID(t *testing.T) {
...
+		err = json.Unmarshal(buf.Bytes(), &response)
+		require.NoError(t, err)
+
+		// verify response
+		require.NotEmpty(t, response)
+		require.Equal(t, 1, len(response.Result))
+	})
+}
+
+func lookupHandler(t *testing.T, op *Operation, path, method string) rest.Handler {
```
Comment (rolsonquadras) [inline pkg/controller/rest/vdri/operation_test.go:265]: may be we can make existing lookup function(line 280) to generic rather than writing new one.

Comment (talwinder50) [reply pkg/controller/rest/vdri/operation_test.go:263]: done

File: pkg/controller/rest/vdri/operation.go (hunk 1 above)
Comment (talwinder50) [reply pkg/controller/rest/vdri/operation.go]: done

File: pkg/controller/rest/vdri/operation_test.go (hunk 2 above)
Comment (talwinder50) [reply pkg/controller/rest/vdri/operation_test.go]: done

File: pkg/controller/rest/vdri/models.go [hunk 4]
```diff
@@ -32,3 +35,55 @@ type createPublicDIDResponse struct {
 	// in: body
 	DID did.Doc `json:"did"`
 }
+
+// emptyRes model
+//
+// swagger:response emptyRes
+type emptyRes struct {
```
Comment (sudeshrshetty) [inline pkg/controller/rest/vdri/models.go:42]: why we need it?

File: pkg/controller/rest/vdri/operation.go [hunk 5]
```diff
@@ -81,6 +91,49 @@ func (o *Operation) CreatePublicDID(rw http.ResponseWriter, req *http.Request) {
 	rest.Execute(o.command.CreatePublicDID, rw, bytes.NewReader(reqBytes))
 }
 
+// SaveDID swagger:route POST /vdri/did vdri saveDIDReq
+//
+// Saves a did document with the friendly name.
+//
+// Responses:
+//    default: genericError
+//        200: emptyRes
```
Comment (sudeshrshetty) [inline pkg/controller/rest/vdri/operation.go:100]: no need of empty response, swagger takes care of it if none specified

File: pkg/controller/rest/vdri/operation.go [hunk 6]
```diff
@@ -81,6 +91,49 @@ func (o *Operation) CreatePublicDID(rw http.ResponseWriter, req *http.Request) {
...
+//
+// Responses:
+//    default: genericError
+//        200: documentRes
+func (o *Operation) GetDID(rw http.ResponseWriter, req *http.Request) {
+	id := mux.Vars(req)["id"]
+
+	decodedID, err := base64.StdEncoding.DecodeString(id)
+	if err != nil {
+		rest.SendHTTPStatusError(rw, http.StatusBadRequest, vdri.InvalidRequestErrorCode, err)
```
Comment (sudeshrshetty) [inline pkg/controller/rest/vdri/operation.go:117]: may be wrap error with some message? just to let caller know where it went wrong, something like "invalid id"

File: pkg/controller/rest/vdri/operation.go (hunk 5 above)
Comment (talwinder50) [reply pkg/controller/rest/vdri/operation.go]: done

File: pkg/controller/rest/vdri/operation.go (hunk 6 above)
Comment (talwinder50) [reply pkg/controller/rest/vdri/operation.go]: done

File: pkg/controller/rest/vdri/operation.go [hunk 7]
```diff
@@ -81,6 +91,48 @@ func (o *Operation) CreatePublicDID(rw http.ResponseWriter, req *http.Request) {
...
+//
+// Responses:
+//    default: genericError
+//        200: documentRes
+func (o *Operation) GetDID(rw http.ResponseWriter, req *http.Request) {
+	id := mux.Vars(req)["id"]
+
+	decodedID, err := base64.StdEncoding.DecodeString(id)
+	if err != nil {
+		rest.SendHTTPStatusError(rw, http.StatusBadRequest, vdri.InvalidRequestErrorCode, fmt.Errorf("invalid id"))
```
Comment (sudeshrshetty) [inline pkg/controller/rest/vdri/operation.go:116]: wrap error with invalid id

**Pull request merged**